server-common/src/main/java/org/apache/kafka/server/common/TestFeatureVersion.java

```diff
@@ -21,16 +21,17 @@
 
 public enum TestFeatureVersion implements FeatureVersion {
 
-    TEST_0(0, MetadataVersion.IBP_3_3_IV0, Collections.emptyMap()),
+    // TEST_1 released right before MV 3.7-IVO was released, and it has no dependencies
     TEST_1(1, MetadataVersion.IBP_3_7_IV0, Collections.emptyMap()),
+    // TEST_2 released right before MV 3.8-IVO was released, and it depends on this metadata version
     TEST_2(2, MetadataVersion.IBP_3_8_IV0, Collections.singletonMap(MetadataVersion.FEATURE_NAME, MetadataVersion.IBP_3_8_IV0.featureLevel()));
 
-    private short featureLevel;
-    private MetadataVersion metadataVersionMapping;
-    private Map<String, Short> dependencies;
+    private final short featureLevel;
+    private final MetadataVersion metadataVersionMapping;
+    private final Map<String, Short> dependencies;
 
     public static final String FEATURE_NAME = "test.feature.version";
-    public static final TestFeatureVersion PRODUCTION_VERSION = TEST_1;
+    public static final TestFeatureVersion LATEST_PRODUCTION = TEST_1;
```

Comment:
Does it mean that the broker will show the test feature to the clients?

Reply:
It does not since in Features, I don't define a production version. I set this for testing purposes. But Features.PRODUCTION_VERSIONS does not contain this feature.

Reply:
🤦‍♀️ I realized I fixed this in the next PR. I will fix this here.